src/newgrf_storage.h

```diff
@@ -65,26 +65,10 @@ struct BasePersistentStorageArray {
  */
 template <typename TYPE, uint SIZE>
 struct PersistentStorageArray : BasePersistentStorageArray {
-	TYPE storage[SIZE]; ///< Memory to for the storage array
-	TYPE *prev_storage; ///< Memory to store "old" states so we can revert them on the performance of test cases for commands etc.
+	using StorageType = std::array<TYPE, SIZE>;
 
-	/** Simply construct the array */
-	PersistentStorageArray() : prev_storage(nullptr)
-	{
-		memset(this->storage, 0, sizeof(this->storage));
-	}
-
-	/** And free all data related to it */
-	~PersistentStorageArray()
-	{
-		free(this->prev_storage);
-	}
-
-	/** Resets all values to zero. */
-	void ResetToZero()
-	{
-		memset(this->storage, 0, sizeof(this->storage));
-	}
+	StorageType storage{}; ///< Memory for the storage array
+	std::unique_ptr<StorageType> prev_storage{}; ///< Memory to store "old" states so we can revert them on the performance of test cases for commands etc.
```

Comment:
"on the performance of test cases" .. it took me a while to parse that sentence :D Maybe reword it a bit to read a bit more natural? "so we can revert after running commands in test-mode" or something? Dunno, but "performance" threw me off here.

Reply:
Not gonna lie, I never read all that comment as I didn't write it either...